File: 프로그래머스/2/62048. 멀쩡한 사각형/멀쩡한 사각형.cpp
#include <cmath>

using namespace std;

int GetGCD(int a, int b){
    if(a % b == 0){
        return b;
    }
    return GetGCD(b, a % b);
}

long long solution(int w,int h) {
    long long answer = 1;
    
    int gcd = GetGCD(w, h);
    
    long long minW = (long long)w / gcd;
    long long minH = (long long)h / gcd;
    
    int deleted = minW * minH - (minW - 1) * (minH - 1);
    
    return (long long)w * h - (deleted * gcd);
    return answer;
}

File: 프로그래머스/2/62048. 멀쩡한 사각형/멀쩡한 사각형.java
class Solution {
    public long solution(int w, int h) {
        int gcb = gcb(w, h);
        int ww = w / gcb;
        int hh = h / gcb;

        return (long) w * h - ((long) (ww + hh - 1) * gcb);
    }

    private int gcb(int a, int b) {
        while (b > 0) {
            int temp = a % b;
            a = b;
            b = temp;
        }
        return a;
    }
}